src/nominatim_api/localization.py

```diff
@@ -7,26 +7,77 @@
 """
 Helper functions for localizing names of results.
 """
-from typing import Mapping, List, Optional
-
 import re
+from typing import List, Mapping, Optional
+
+from .config import Configuration
+
 
 class Locales:
     """ Helper class for localization of names.
 
-        It takes a list of language prefixes in their order of preferred
-        usage.
+    It takes a list of language prefixes in their order of preferred
+    usage.
+    It takes config object as input which enables to configure the 
+    list of supported tags. Setting name: `OUTPUT_NAMES`
+    How to use?
+    e.g. `name:XX,name,brand,official_name:XX,short_name:XX,official_name,short_name,ref`
+        - Coma separated
+        - `:XX` identifies lang tags
+        - consecutive `:XX` identifies multiple lang tags to-be added at the same time
+        - subsequent parts after lang tags (till next lang tag) identifies batch of tags
     """
 
-    def __init__(self, langs: Optional[List[str]] = None):
+    def __init__(self, langs: Optional[List[str]] = None, config: Optional[Configuration] = None):
         self.languages = langs or []
         self.name_tags: List[str] = []
 
-        # Build the list of supported tags. It is currently hard-coded.
-        self._add_lang_tags('name')
-        self._add_tags('name', 'brand')
-        self._add_lang_tags('official_name', 'short_name')
-        self._add_tags('official_name', 'short_name', 'ref')
+        # Build the list of supported tags
+        # Uses hard-coded when config `OUTPUT_NAMES` not provided
+        if config and config.OUTPUT_NAMES != '':
+            self._build_output_name_tags(config.OUTPUT_NAMES)
+        else:
+            self._build_default_output_name_tags()
+
+
+    def _build_default_output_name_tags(self) -> None:
+        """
+        Build the list of supported tags. Hard-coded implementation.
+        """
+        self._add_lang_tags("name")
+        self._add_tags("name", "brand")
+        self._add_lang_tags("official_name", "short_name")
+        self._add_tags("official_name", "short_name", "ref")
+
+
+    def _build_output_name_tags(self, nominatim_output_names: str) -> None:
+        """
+        Build the list of supported tags. Dynamic implementation.
+        How to use?
+        e.g. `name:XX,name,brand,official_name:XX,short_name:XX,official_name,short_name,ref`
+            - Coma separated
+            - `:XX` identifies lang tags
+            - consecutive `:XX` identifies multiple lang tags to-be added at the same time
+            - subsequent parts after lang tags (till next lang tag) identifies batch of tags
+        """
+        nominatim_output_names = nominatim_output_names.split(",")
+        lang_tags = []
+        tags = []
```

Comment:
This seems wrong. Order matters here. The name tag list must be in exactly the same order as the input list.

Reply:
I need more context on this. These variables are initialized to build the tags in order as specified in the format, they work in batches.

consider default: `name:XX,name,brand,official_name:XX,short_name:XX,official_name,short_name,ref`

1. for `name:XX` add "name" to `lang_tags`
2. for `name` add "name" to `tags`, check if `lang_tags` has elements then call `_add_lang_tags(*lang_tags)` and initialize `lang_tags = []` again.

In this way tags are processed in batches